src/model/kubernetes.js

```diff
@@ -71,19 +71,21 @@ class Kubernetes {
     await this.kubeClient.api.v1
       .namespaces('default')
       .persistentvolumeclaims.post({ body: pvcManifest });
-    let ready = false;
-    while (!ready) {
-      // eslint-disable-next-line no-await-in-loop
-      const queryResult = await this.kubeClient.api.v1
-        .namespaces('default')
-        .persistentvolumeclaims(this.pvcName)
-        .get();
-      ready = queryResult.body.status.phase !== 'Pending';
+    await Kubernetes.watchBuildJobUntilFinished();
+    core.info('Persistent Volume ready for claims');
+  }
+
+  static async watchPersistentVolumeClaimUntilReady() {
+    const queryResult = await this.kubeClient.api.v1
+      .namespaces('default')
+      .persistentvolumeclaims(this.pvcName)
+      .get();
+    if (queryResult.body.status.phase === 'Pending') {
+      await Kubernetes.watchPersistentVolumeClaimUntilReady();
```

Comment:
This looks much simpler to read indeed.

Now i'm realising also that we're probably hammering the API. Should we put an interval delay on this?